### DISCUSSION TITLE
A proper way to use makeInviteSession to registered UA ?

### DISCUSSION THREAD

**Discussion by pbalash0v** [resiprocate/resiprocate#289] [Q&A] [answered]
I have some UA, that registers with custom reSIProcate-based SIP Registrar. I retrieve `Contact` field from UA's REGISTER request of a kind `<sip:callee@192.168.1.1:52384;transport=TCP>` and store in some container, mapping AoR (let it be `callee@domain.com`) to this `device-URI` for later use. Now I'm willing to create an outgoing INVITE session to that registered UA. I assume the API I'm looking for is `makeInviteSession`  overload methods set in DUM. All relevant methods take a `NameAddr` as a first parameter. 
So I'm assume something along the lines:
```c++
...
auto aor_str = std::string{"callee@domain.com"};
auto stored_contact_string = get_contact_by_aor(aor_str);
auto to = resip::NameAddr{resip::Data{stored_contact_string}};
m_dum.makeInviteSession(to,...);
...
```
This gives SIP message:
```
INVITE sip:callee@192.168.1.1:52384;transport=TCP SIP/2.0
...
To: <sip:callee@192.168.1.1:52384;transport=TCP>
...
```

So the RLine of a resulting INVITE looks ok, but the `To` field is not exactly what I expect to see, given (https://www.rfc-editor.org/rfc/rfc3261#section-8.1.1.2):
```
The To header field first and foremost specifies the desired "logical" recipient of the request, or the address-of-record of the user or resource that is the target of this request.
```

What would be the proper API to achive SIP message, that looks something like this
```
INVITE sip:callee@192.168.1.1:52384;transport=TCP SIP/2.0
...
To: <sip:callee@domain.com>
...
```
?
(I suppose one way to go is to modify `msg->header(resip::h_To).uri()` in result of `makeInviteSession` manually)

Thanks.

**Comment by sgodin** [comment, marked as answer]
You can use the AOR (To Header) in the makeInviteRequest, then modify the
RequestUri of the returned Invite message with the Contact header from the
registration before calling send to send it out.

Scott
